onnxruntime/core/providers/mkldnn/subgraph/mkl_kernel.h

```diff
@@ -0,0 +1,178 @@
+// Copyright(C) 2019 Intel Corporation
+// Licensed under the MIT License
+
+#pragma once
+#ifdef _WIN32
+#pragma warning(disable : 4244)
+#endif
+
+#include "mkldnn.hpp"
+#include "core/common/cpuid_info.h"
+#include "core/providers/mkldnn/subgraph/subgraph.h"
+
+namespace onnxruntime {
+namespace mkl_dnn {
+
+class MklKernel {
+ public:
+  explicit MklKernel(MklNode& node,
+                     MKLDNNExecutionProvider* provider,
+                     std::shared_ptr<MKLContext> mkl_context) {
+    mkl_context_ = mkl_context;
+    mklnode_ptr_ = std::make_shared<MklNode>(node);
+    provider_ = provider;
+    alloc_ = provider_->GetAllocator(0, OrtMemTypeDefault);
+  }
+  virtual ~MklKernel(){};
+
+  virtual void ReadAttributes(const std::unordered_map<std::string,
```

Comment:
is it possible to leverage OpNodeProtoHelper for reading attributes? 
then you wouldn't need to define all the Get*Attr() methods below.

Reply:
I looked into it and it's not straight forward. Can this be done after this PR?